lib/packer/builders/amazon.rb

```diff
@@ -12,7 +12,7 @@ def initialize
           'instance_type',
           'region',
           'source_ami',
-          'ssh_username'
+          ['ssh_username', 'communicator']
```

Comment:
That sounds like a good idea to me. I think I would have done that if I was
a little more confident haha. I've never worked on a ruby project with
someone who has more experience than me (which is not very much). Thanks!

I was looking at your lowered expectations gem and it seems like a really
cool idea... I think this gem and your todo with the packer validation
command would have caught a problem I just had with packer being out of
date on a specific machine and not liking my communicator fields. I might
try to work on this and send you another pull request later if you're
interested. Thanks for making this project available!

-Greg

On Sun, Sep 13, 2015 at 8:02 PM, Ian Chesal <notifications@github.com>
wrote:

> In lib/packer/builders/amazon.rb
> <https://github.com/ianchesal/packer-config/pull/10#discussion_r39354271>:
>
> > @@ -12,7 +12,7 @@ def initialize
> >            'instance_type',
> >            'region',
> >            'source_ami',
> > -          'ssh_username'
> > +          ['ssh_username', 'communicator']
>
> Ah. Gotcha. I see what you mean. Yea, I can make this work. It looks like
> the communicator stuff should get pulled out in to a module that can get
> dropped in to multiple builder classes. Shouldn't be too hard!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ianchesal/packer-config/pull/10/files#r39354271>.
>